src/restconfig/src/main/java/org/geoserver/rest/security/AuthenticationFilterChainRestController.java

```diff
@@ -5,121 +5,647 @@
 public class AuthenticationFilterChainRestController extends RestBaseController {
+
     private final GeoServerSecurityManager securityManager;
 
     public AuthenticationFilterChainRestController(GeoServerSecurityManager securityManager) {
         this.securityManager = securityManager;
     }
 
-    @GetMapping(
-            produces = {
-                MediaType.APPLICATION_JSON_VALUE,
-                MediaType.APPLICATION_XML_VALUE,
-            })
-    public RestWrapper<AuthFilterChain> list() {
-        checkAuthorisation();
-        List<AuthFilterChain> filterChains = listFilterChains();
-        return wrapList(filterChains, AuthFilterChain.class);
+    // ---------- XStream wiring ----------
+    private static void configureAliases(XStreamPersister persister) {
+        XStream xs = persister.getXStream();
+
+        xs.allowTypesByWildcard(new String[] {"org.geoserver.rest.security.xml.*"});
+
+        // root
+        xs.alias("filterChain", AuthFilterChainCollection.class);
+
+        // collection: no <chains> wrapper; each item is <filters>
+        xs.addImplicitCollection(AuthFilterChainCollection.class, "chains", "filters", AuthFilterChainFilters.class);
+
+        // item
+        xs.alias("filters", AuthFilterChainFilters.class);
+        xs.aliasField("class", AuthFilterChainFilters.class, "clazz");
+        xs.aliasAttribute(AuthFilterChainFilters.class, "requireSSL", "ssl");
+        // Disable XStream's special meaning for the "class" attribute
+        xs.aliasSystemAttribute(null, "class");
+        xs.aliasSystemAttribute(null, "resolves-to"); // (optional, belts & braces)
+
+        xs.useAttributeFor(AuthFilterChainFilters.class, "name");
+        xs.useAttributeFor(AuthFilterChainFilters.class, "clazz");
+        xs.useAttributeFor(AuthFilterChainFilters.class, "path");
+        xs.useAttributeFor(AuthFilterChainFilters.class, "disabled");
+        xs.useAttributeFor(AuthFilterChainFilters.class, "allowSessionCreation");
+        xs.useAttributeFor(AuthFilterChainFilters.class, "requireSSL");
+        xs.useAttributeFor(AuthFilterChainFilters.class, "matchHTTPMethod");
+        xs.useAttributeFor(AuthFilterChainFilters.class, "interceptorName");
+        xs.useAttributeFor(AuthFilterChainFilters.class, "exceptionTranslationName");
+        xs.useAttributeFor(AuthFilterChainFilters.class, "roleFilterName");
+
+        xs.addImplicitCollection(AuthFilterChainFilters.class, "filters", "filter", String.class);
     }
 
+    // ---------- Endpoints ----------
+
     @GetMapping(
-            value = "/{chainName}",
-            produces = {
-                MediaType.APPLICATION_JSON_VALUE,
-                MediaType.APPLICATION_XML_VALUE,
-            })
-    public RestWrapper<AuthFilterChain> view(@PathVariable("chainName") String chainName) {
-        checkAuthorisation();
+            path = "",
+            produces = {MediaType.APPLICATION_XML_VALUE, MediaType.APPLICATION_JSON_VALUE})
```

Comment:
Hum... still stating it supports JSON too. And it should be able to, like the rest of the REST API (JSON is not the best format, but having a REST API that does not support JSON is some parts is even more confusing IMHO).